packages/devtools_app/lib/src/debugger/debugger_model.dart

```diff
@@ -702,18 +751,63 @@ class Variable extends TreeNode<Variable> {
     return Variable._(null, null, text);
   }
 
+  factory Variable.grouping(
+    GenericInstanceRef ref, {
+    @required int offset,
+    @required int count,
+  }) {
+    return Variable._(
+      null,
+      ref,
+      '[$offset - ${offset + count - 1}]',
+      offset: offset,
+      childCount: count,
+    );
+  }
+
+  static const MAX_CHILDREN_IN_GROUPING = 100;
+
   final String text;
   final String name;
   GenericInstanceRef get ref => _ref;
   GenericInstanceRef _ref;
 
+  int get offset {
+    if (_offset != null) return _offset;
+    return 0;
+  }
+
+  int _offset;
+
+  int get childCount {
+    if (_childCount != null) return _childCount;
+
+    final value = this.value;
+    if (value is InstanceRef) {
+      if (value.kind == InstanceKind.kList) {
+        return value.length ?? 0;
+      } else if (value.kind == InstanceKind.kMap) {
+        return value.length ?? 0;
+      } else if (value.kind != null && value.kind.endsWith('List')) {
+        // Catch-all for Uint8ClampedList, Uint8List, Uint16List, Uint32List,
+        // Uint64List, Int8List, Int16List, Int32List, Int64List, Flooat32List,
+        // Float64List, Inst32x3List, Float32x4List, and Float64x2List types:
+        return value.length ?? 0;
```

Comment:
can you combine the above three checks into one if check since they all return the same thing?

Reply:
Done, thanks!